### PR TITLE
Adjusted recipe argument parsing

### PR DESCRIPTION
**Pull request by ramo-j**
Adjusted argument parsing from recipes to use a new dataclass. Change is transparent to a user, but is prep work for #702

### CODE REVIEW

File: dftimewolf/lib/recipes/manager.py [hunk 1]
```diff
@@ -35,7 +35,12 @@ def _ReadRecipeFromFileObject(
     description = json_dict['description']
     del json_dict['description']
 
-    args = json_dict['args']
+    args = []
+    for arg_list in json_dict['args']:
+      arg = RecipeArgs(switch=arg_list[0],
+                             help_text=arg_list[1],
```
Comment (sydp) [inline dftimewolf/lib/recipes/manager.py:41]: fix indentation for the next two lines

**Review by sydp — DISMISSED** [review]
One minor style nit but otherwise LGTM

File: dftimewolf/lib/recipes/manager.py (hunk 1 above)
Comment (sydp) [reply dftimewolf/lib/recipes/manager.py:41]: or  you may be able to unpack from the list directly into the dataclass

Comment (ramo-j) [reply dftimewolf/lib/recipes/manager.py]: Done. Went with just fixing the indentation, as I know there's going to be an optional element to this dataclass in a later PR, I can look into your other suggestion.

**Review by sydp — APPROVED** [review]
LGTM

**Pull request merged**